### PR TITLE
Docs improvement

### PR DESCRIPTION
**Pull request by Pratosh22**
-Added IconButton docs
-Added docs for Toast promise
-Fixed dropdown bug in docs

### CODE REVIEW

File: packages/docs/docs/components/dropdown.mdx [hunk 1]
```diff
@@ -16,10 +16,16 @@ import { DropdownMenu,
 ### Usage 
 ```jsx live
 <DropdownMenu>
-  <DropdownMenuTrigger asChild>
-    <Button aria-label="Customise options">
-      More Options
-    </Button>
+  <DropdownMenuTrigger style={{
```
Comment (maneeshp97) [inline packages/docs/docs/components/dropdown.mdx:19]: @Pratosh22 Why are we using style attribute here ? Can't we use CSS prop ?

File: packages/docs/docs/components/icon-button.mdx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  <IconButton icon={<EmailIcon />} />
+  <IconButton icon={<DownloadIcon />} />
+</Flex>
+```
+
+### Props
+
+| Property | Description                                | Type    | Default | Options                                                       |
+| -------- | ------------------------------------------ | ------- | ------- | ------------------------------------------------------------- |
+| size     | Size of the icon button                    | string  | -       | 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', '2xl'                    |
```
Comment (maneeshp97) [inline packages/docs/docs/components/icon-button.mdx:24]: @Pratosh22 what about the `icon` prop?

File: packages/docs/docs/components/icon-button.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  <IconButton icon={<DownloadIcon />} />
+</Flex>
+```
+
+### Props
+
+| Property | Description                                | Type    | Default | Options                                                       |
+| -------- | ------------------------------------------ | ------- | ------- | ------------------------------------------------------------- |
+| icon     | Icon to be rendered within the button      | node    | -       | -                                                             |
+| size     | Size of the icon button                    | string  | -       | 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', '2xl'                    |
```
Comment (akzhysparrow) [inline packages/docs/docs/components/icon-button.mdx:25]: Can you add the default values for size, color and variant? @Pratosh22

File: packages/docs/docs/components/toast.mdx [hunk 4]
```diff
@@ -45,7 +45,35 @@ Using `toast` hook you can trigger toast from anywhere in your application.
...
 ```
 
+### Toast With Promise
+Using `toast.promise` you can show toast with promise.
 
+```jsx
+<Button
+variant="outline"
+onClick={() => {
+  //highlight-start
```
Comment (akzhysparrow) [inline packages/docs/docs/components/toast.mdx:55]: Is this comment required? Is code highlighting not working without this? Also can you prettify this block?

Comment (Pratosh22) [reply packages/docs/docs/components/toast.mdx:55]: Yes, we need the comment to highlight the code @akzhysparrow

**Review by akzhysparrow — APPROVED** [review]

**Review by maneeshp97 — APPROVED** [review]

**Review by maneeshp97 — APPROVED** [review]

**Pull request merged**